src/job.ts

```diff
@@ -79,11 +79,12 @@ export class Job {
     readonly environment?: {name: string; url: string | null; deployment_tier: string | null; action: string | null};
     readonly jobId: number;
     readonly rules?: {
-        if: string;
-        when: string;
-        exists: string[];
-        allow_failure: boolean;
-        variables: {[key: string]: string};
+        if?: string;
+        when?: string;
+        changes?: string[] | {paths: string[]};
+        exists?: string[];
+        allow_failure?: boolean;
+        variables?: {[name: string]: string};
```

Comment:
I changed this to match https://github.com/firecow/gitlab-ci-local/blob/70d2462f8b00fb74e902ca33ed87cdcaee6b083a/src/utils.ts#L16-L23

Reply:
lets export this 

ie. 

```typescript
export type JobRule = {
    if?: string;
    when?: string;
    changes?: string[] | {paths: string[]};
    exists?: string[];
    allow_failure?: boolean;
    variables?: {[name: string]: string};
};
```
then import it in src/utils

so that we don't need to sync the types in two places